llama/sampling_ext.cpp

```diff
@@ -68,21 +69,65 @@ int schema_to_grammar(const char *json_schema, char *grammar, size_t max_len)
     }
 }
 
-struct llama_vocab * llama_load_vocab_from_file(const char * fname) {
-    llama_vocab * vocab = new llama_vocab();
+
+struct grammar *grammar_init(char* grammar) {
```

Comment:
This seemed like the simplest way to carve out the grammar. Open to naming suggestions ik we're not a fan of `New` and `Init`